NICE.Identity.Test/UnitTests/Authorisation.WebAPI/Services/EmailServiceTests.cs

```diff
@@ -0,0 +1,219 @@
+{
+    public class EmailServiceTests : TestBase
+    {
+
+        private readonly Mock<ILogger<EmailService>> _emailServiceLogger;
+        private readonly MockWebHostEnvironment _webHostEnvironment;
+        private readonly SmtpClient _smtpClient;
+        private readonly int _localSmtpPort;
+
+        public EmailServiceTests()
+        {
+            _emailServiceLogger = new Mock<ILogger<EmailService>>();
+            _webHostEnvironment = new MockWebHostEnvironment();
+            _smtpClient = new SmtpClient();
+
+            _localSmtpPort = 3333;
+
+            Identity.Authorisation.WebAPI.Configuration.AppSettings.EmailConfig.Server = "localhost";
+            Identity.Authorisation.WebAPI.Configuration.AppSettings.EmailConfig.Port = _localSmtpPort;
+            Identity.Authorisation.WebAPI.Configuration.AppSettings.EmailConfig.SenderAddress = "sender@example.com";
+
+        }
+
+        [Fact]
+        public void test_single_email()
+        {
+            //Arrange
+            var emailService = new EmailService(_webHostEnvironment, _emailServiceLogger.Object, _smtpClient);
+            var user = new User() { EmailAddress = "address@example.com" };
+
+            using var emailServer = netDumbster.smtp.SimpleSmtpServer.Start(_localSmtpPort);
+
+            //Act
+            emailService.SendEmail<TestEmailGenerator>(user);
+
+            //Assert
+            var email = emailServer.ReceivedEmail.Single();
+
+            email.FromAddress.ToString().ShouldBe("sender@example.com");
+            email.ToAddresses.Single().ToString().ShouldBe("address@example.com");
+
+            email.MessageParts
+                .Single(x => x.HeaderData.Contains("text/plain"))
+                .BodyData
+                .ShouldBe("Title - EmailTitle; Body - TextEmailBody");
+
+            email.MessageParts
+                .Single(x => x.HeaderData.Contains("text/html"))
+                .BodyData
+                .ShouldBe("<html>\r\n<head>\r\n    <Title>EmailTitle</Title>\r\n</head>\r\n<body>\r\n    HtmlEmailBody\r\n    <a href=\"mailto:contactus@example.com?subject=Test%20contact%20us%20subject\">Contact Us</a>\r\n</body>\r\n</html>");
+        }
+
+        [Fact]
+        public void send_pending_registration_deleted_email()
+        {
+            //Arrange
+            var emailService = new EmailService(_webHostEnvironment, _emailServiceLogger.Object, _smtpClient);
+            
+            var users = new List<User>() {
+                new User {
+                    NameIdentifier = "auth|userOne",
+                    EmailAddress = "UserOne@example.com" },
+                new User {
+                    NameIdentifier = "auth|userTwo",
+                    EmailAddress = "UserTwo@example.com" },
+            };
+
+            using var emailServer = netDumbster.smtp.SimpleSmtpServer.Start(_localSmtpPort);
+            //Act
+            emailService.SendPendingRegistrationDeletedEmail(users);
```

Comment:
Can this be indented to show the scope of the using statement. Same with the other using's in this file

Reply:
I'm not sure about this one, I had a go at using these braceless "using" statements. I think that the idea is that the variable lives until the end of the scope but has Dispose called immediately when the scope closes. I tested without the "using" and it looked like the object stayed alive once the scope had closed. I get what you are saying, but indenting also felt wrong.

Either way I'm not sold on this form of using statement so I went back and put the braces in. I'm just a sucker for a visual studio code cleanup hint. for better or for worse.